source/run-command/run-command.rst

```diff
@@ -0,0 +1,185 @@
+
+      /**
+       * An optional explicit client session.
+       * The associated logical session id (`lsid`) MUST be applied to the command.
+       *
+       * @see https://www.mongodb.com/docs/manual/core/read-isolation-consistency-recency/#std-label-sessions
+       */
+      session?: ClientSession;
+
+      /**
+       * An optional timeout option to govern the amount of time that a single operation can execute before control is returned to the user.
+       * This timeout applies to all of the work done to execute the operation, including but not limited to server selection, connection checkout, and server-side execution.
+       *
+       * @ see https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst
+       */
+      timeoutMS?: number;
+    }
+
+Deviations
+^^^^^^^^^^
+
+* Drivers MAY rename their API to match the language syntax.
+
+RunCommand implementation details
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+RunCommand provides a way to access MongoDB server commands directly without requiring a driver to implement a bespoke helper.
+The API is intended to take a document from a user and apply a number of common driver internal concerns before forwarding the command to a server.
+In general, a driver SHOULD avoid inspecting or modifying the user's command document in an effort to remain forward compatible with any potential server commands.
+
+Drivers that have historically modified user input SHOULD strive to instead clone the input such that appended fields do not affect the user's input.
+
+OP_MSG
+""""""
+
+The ``$db`` global command argument MUST be sent on the command sent to the server and it MUST correspond to the database name the runCommand API was invoked on.
+
+* See OP_MSG's section on `Global Command Arguments <https://github.com/mongodb/specifications/blob/master/source/message/OP_MSG.rst#global-command-arguments>`_
+
+ReadPreference
+""""""""""""""
+
+A driver's runCommand implementation MUST assume all commands are read operations.
+To facilitate selecting the server the command should be sent to a driver's runCommand MUST accept an optional Read Preference option.
+
+* See Server Selection's section on `Use of read preferences with commands <https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#use-of-read-preferences-with-commands>`_
+
+If the provided Read Preference is NOT primary, the command sent MUST include the ``$readPreference`` global command argument.
+
+* See OP_MSG's section on `Global Command Arguments <https://github.com/mongodb/specifications/blob/master/source/message/OP_MSG.rst#global-command-arguments>`_
+
+Driver Sessions
+"""""""""""""""
+
+A driver's runCommand API MUST provide an optional session option to support explicit sessions and transactions.
+If a session is not provided the driver MUST attach an implicit session if the selected server supports sessions.
```

Comment:
Is this still relevant if drivers now assume all servers support sessions? IIRC, min supported server version is now 3.6. I think this came up in another spec, but I could be mistaken.

Reply:
Mongocryptd does not support sessions, the recent change to the spec related to this was that drivers no longer perform server selection to determine session support, but support for sessions is still checked before attaching an lsid